rclcpp_components/src/node_main.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "class_loader/class_loader.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_components/node_factory.hpp"
+
+
+#define LINKTIME_COMPOSITION_LOGGER_NAME "node_main"
+
+int main(int argc, char * argv[])
+{
+  // Force flush of the stdout buffer.
+  setvbuf(stdout, NULL, _IONBF, BUFSIZ);
```

Comment:
I don't know if we want to make this decision for everyone using components. We basically do it in our examples to ensure that while testing output comes immediately. But for instance, Python does not do this automatically and instead you have to turn it on with an environment variable or command line argument. Maybe that could be done in this case as well (have a command line argument for it that this main function can use.